core/src/Kokkos_Macros.hpp

```diff
@@ -382,6 +382,10 @@
 #define KOKKOS_IMPL_DEVICE_FUNCTION
 #endif
 
+#if !defined(KOKKOS_IMPL_PRINTF)
+#define KOKKOS_IMPL_PRINTF(...) printf(__VA_ARGS__)
```

Comment:
I would add a comment that this macro is temporary and will disappear as soon as SYCL supports `printf` in kernels.